common/app/views/fragments/newHeader.scala.html

```diff
@@ -26,16 +23,17 @@ <h1 class="u-h">The Guardian</h1>
         @fragments.nav.editionPicker()
 
         <nav class="new-header__nav">
-            <a href="#main-menu" class="new-header__nav__menu-button js-change-link" tabindex="0">
-                <i class="new-header__veggie-burger-icon">
-                </i>
+            @NewNavigation.topLevelSections.map { section =>
+                @primaryLinks(section.name)
+            }
+            <label for="main-menu-toggle" class="new-header__nav__menu-button js-change-link" tabindex="0">
+                <i class="new-header__veggie-burger-icon"></i>
```

Comment:
The icon is pure HTML / CSS, which allows us to animate the elements using CSS transform and transition, magically transforming the three-line veggie burger into a X icon. 

I do agree that `<span>` is semantically more correct than `<i>`, so I will update.